Configurations/50-win-onecore.conf

```diff
@@ -0,0 +1,56 @@
+# Windows OneCore targets.
+#
+# OneCore is new API stability "contract" that transends Desktop, IoT and
+# Mobile[?] Windows editions. It's a set up "umbrella" libraries that
+# export subset of Win32 API that are common to all Windows 10 devices.
+#
+# TODO: drop onecore_downlevel.lib.
+
+my %targets = (
+    "VC-WIN32-ONECORE" => {
+        inherit_from    => [ "VC-WIN32" ],
+        # /NODEFAULTLIB:kernel32.lib is needed, because MSVCRT.LIB has
+        # hidden reference to kernel32.lib, but we don't actually want
+        # it in "onecore" build.
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+    },
+    "VC-WIN64A-ONECORE" => {
+        inherit_from    => [ "VC-WIN64A" ],
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+    },
+
+    # Windows on ARM targets. ARM compilers are additional components in
+    # VS2017, i.e. they are not installed by default. And when installed,
+    # there are no "ARM Tool Command Prompt"s on Start menu, you have
+    # to locate vcvarsall.bat and act accordingly. VC-WIN32-ARM has
+    # received limited testing with evp_test.exe on Windows 10 IoT Core,
+    # but not VC-WIN64-ARM, no hardware... In other words they are not
+    # actually supported...
+    #
+    # Another thing to keep in mind [in cross-compilation scenario such
+    # as this one] is that target's file system has nothing to do with
+    # compilation system's one. This means that you're are likely to use
+    # --prefix and --openssldir with target-specific values. 'nmake install'
+    # step is effectively meaningless in cross-compilation case, though
+    # it might be useful to 'nmake install DESTDIR=S:\ome\where' where you
+    # can point Visual Studio to when compiling custom application code.
+
+    "VC-WIN32-ARM" => {
+        inherit_from    => [ "VC-noCE-common" ],
+        defines         => add("_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE"),
+        bn_ops          => "BN_LLONG RC4_CHAR EXPORT_VAR_AS_FN",
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+        multilib        => "-arm",
+    },
+    "VC-WIN64-ARM" => {
+        inherit_from    => [ "VC-noCE-common" ],
+        defines         => add("_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE"),
+        bn_ops          => "SIXTY_FOUR_BIT RC4_CHAR EXPORT_VAR_AS_FN",
+        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        ex_libs         => "onecore.lib onecore_downlevel.lib",
+        multilib        => "-arm64",
+    },
+);
```

Comment:
You should add those config target names in [`INSTALL`](https://github.com/openssl/openssl/blob/master/INSTALL#L95)

Reply:
If comment is about ARM targets, then I'd be against, because they are not actually supported. As commentary mentions, ARM64 is not tested *at all*, and ARM has received very limited [low-level algorithm] testing. If comment applies even to ONECORE, I'd say let's get rid of onecore_downlevel first. As it is now, it's really work in progress. On related side note one can wonder is it actually needed to list *all* VC targets in INSTALL? We don't list all Linux, BSD, Solaris, HP-UX ones :-) :-) :-)

Reply:
Note that for the Unix flavored platforms, we document the use of `./config`, which automates discovery.  For Windows, we don't, and instead list the available config targets.  I see nothing weird with that...

So why don't we at least list those with expect people to start using?  From listening to the conversation on Onecore, I suspect those will become popular config targets pretty soon, so I do think those should be listed at the very least.

Reply:
> So why don't we at least list those with expect people to start using? From listening to the conversation on Onecore, I suspect those will become popular config targets pretty soon, 

Same conversation suggests that it won't unless we drop onecore_downlevel :-) Once again, as it stands now, Onecore is work in progress, and will remain work in progress after this commit. And suggestion is to add new targets to INSTALL *after* we omit onecore_downlevel.

Reply:
Okie, I'm fine with that.  Let's just not forget it.